src/main/scala/analysis/solvers/FixPointSolver.scala

```diff
@@ -261,25 +274,21 @@ trait SimplePushDownWorklistFixpointSolver[N, T, L <: Lattice[T]] extends PushDo
     run(domain)
     x
 
-trait WorklistFixPointFunctions[N, T, L <: Lattice[T]]  extends  LinkedHashSetWorklist[N]:
 
-  val lattice: MapLattice[N, T, L]
-
-  var x: Map[N, T]
+/**
+ * Push down worklist solver that maps first to init
+ *
+ * @tparam N
+ *   type of the elements in the worklist.
+ * @tparam T Type of the elements in the sublattice
+ * @tparam L Sublattice
+ */
+trait InitializingPushDownWorklistFixpointSolver[N, T, L <: Lattice[T]] extends PushDownWorklistFixPointFunctions[N, T, L]:
```

Comment:
Why doesn't this just extend PushDownWorklistFixPointSolver? This doesn't make sense on its own as it doesn't have a process method. I understand that it is only used by the IDESolver classes which implement their own process methods.

Reply:
PushDownWorklistFixPointSolver extends the `Dependencies` trait with a type restriction N where N is the elements in the worklist. The worklist elements for IDE analysis is a triple/tuple like (CFGPosition, DL, DL)/(CFGPosition, DL). I could write a forward and backward iterator that detuples first and calls the IL iterator. PushDownWorklistFixPointSolver also extends MapLatticeSolver, but the IDESolver phases that use the worklist don't require MapLatticeSolver members

Reply:
ah, the TIP inheritance is such a nightmare. I would like to try to organise this better but it'll take a bit more thought and isn't an urgent issue, so it's fine the way it is for now